lib/components/tiptap/image/image.ts

```diff
@@ -1,103 +1,164 @@
-import { mergeAttributes, nodeInputRule } from '@tiptap/core'
-import { ReactNodeViewRenderer } from '@tiptap/react'
-import { ImageResizeComponent } from './ImageResizeComponent'
-import Image from '@tiptap/extension-image'
+import { mergeAttributes, nodeInputRule } from "@tiptap/core";
+import { ReactNodeViewRenderer } from "@tiptap/react";
+import { ImageResizeComponent } from "./ImageResizeComponent";
+import Image from "@tiptap/extension-image";
 
 export interface ImageOptions {
-  inline: boolean
-  allowBase64: boolean
-  HTMLAttributes: Record<string, any>
-  useFigure: boolean
-  readOnly: boolean
+  inline: boolean;
+  allowBase64: boolean;
+  HTMLAttributes: Record<string, any>;
+  useFigure: boolean;
+  readOnly: boolean;
+  handleImageUpload?: (id: string) => Promise<void>;
+  deleteImage?: (id: string) => Promise<void>;
 }
-declare module '@tiptap/core' {
+
+declare module "@tiptap/core" {
   interface Commands<ReturnType> {
     imageResize: {
       setImage: (options: {
-        src: string
-        alt?: string
-        title?: string
-        width?: string | number
-        height?: string | number
-        isDraggable?: boolean
-      }) => ReturnType
-    }
+        src: string;
+        alt?: string;
+        title?: string;
+        width?: string | number;
+        height?: string | number;
+        isDraggable?: boolean;
+      }) => ReturnType;
+    };
   }
 }
-export const inputRegex = /(!\[(.+|:?)]\((\S+)(?:(?:\s+)["'](\S+)["'])?\))$/
+export const inputRegex = /(!\[(.+|:?)]\((\S+)(?:(?:\s+)["'](\S+)["'])?\))$/;
 export const ImageResize = Image.extend<ImageOptions>({
-  name: 'imageResize',
+  name: "imageResize",
   addOptions() {
     return {
       inline: false,
       allowBase64: false,
       HTMLAttributes: {},
       useFigure: false,
       readOnly: false,
-    }
+      handleImageUpload: undefined,
+      deleteImage: undefined,
+    };
+  },
+  addCommands() {
+    const { deleteImage } = this.options;
+    return {
+      setImage:
+        (options) =>
+        ({ commands }) => {
+          const uniqueId = `image-${Date.now()}-${Math.random()
```

Comment:
Could we instead do `actualFilename-${Date.now()}` only?

You might have to split user's original filename with `.` and discard the last item of array -> join rest to get actualFilename

Reply:
If you use Unix Nano for `Date` the random part should be unnecessary too right?